cmd/gateway-main.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Minio Cloud Storage, (C) 2015, 2016, 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"github.com/gorilla/mux"
+	"github.com/minio/cli"
+)
+
+var gatewayCmd = cli.Command{
+	Name:   "gateway",
+	Usage:  "Start in gateway mode.",
+	Action: gatewayMain,
+	Subcommands: []cli.Command{
+		{
+			Name:  "azure",
+			Usage: "Azure cloud",
```

Comment:
We need a helpTemplate here just like serverMain() to provide examples for how to run azure.. Can you add these examples? 

Also can you create a simple markdown documenting the gateway changes perhaps ideally in docs/gateway directory top-level README.md would do.